src/stores/TransactionStore.tsx

```diff
@@ -0,0 +1,48 @@
+import { TransactionTableData } from '@/components/AddTransactionForm/AddTransactionForm';
+import { showErrorToast, showSuccessToast } from '@/utils/showToast';
+import { create } from 'zustand';
+
+type TransactionStore = {
+  transactions: TransactionTableData[];
+  addTransaction: (transaction: TransactionTableData) => void;
+  editTransaction: (updatedTransaction: TransactionTableData) => void;
+  deleteTransaction: (id: string) => void;
+};
+
+export const useTransactionStore = create<TransactionStore>((set) => ({
+  transactions: JSON.parse(localStorage.getItem('transactions') || '[]'),
+
+  addTransaction: (transaction) =>
+    set((state) => {
+      const newTransactions = [...state.transactions, transaction];
+      localStorage.setItem('transactions', JSON.stringify(newTransactions));
+      return { transactions: newTransactions };
+    }),
+
+  editTransaction: (updatedTransaction) =>
+    set((state) => {
+      const newTransactions = state.transactions.map((transaction) =>
+        transaction.id === updatedTransaction.id
+          ? updatedTransaction
+          : transaction,
+      );
+      localStorage.setItem('transactions', JSON.stringify(newTransactions));
+      return { transactions: newTransactions };
+    }),
+
+  deleteTransaction: (id) => {
+    try {
+      set((state) => {
+        const newTransactions = state.transactions.filter(
+          (transaction) => transaction.id !== id,
+        );
+        localStorage.setItem('transactions', JSON.stringify(newTransactions));
+        return { transactions: newTransactions };
+      });
+      showSuccessToast('Transakce byla úspěšně smazána');
+    } catch (error) {
```

Comment:
Try catch block should contain a least amount of code that necessary. In this case `localStorage.setItem()` is the code that can throw error, that should be the only thing wrapped. Alternatively you can check how error boundaries work, that may eliminate using this try catch block.

Reply:
I've updated the try-catch block to wrap around `localStorage.setItem()` only for now, but I'll also check out error boundaries for future improvements :)